src/Helpers/CreateTokboxPublisher.js

```diff
@@ -58,12 +58,13 @@ const publishToSession = (channelName, publisher, session) => {
         console.log(`Retrying publication (attempt ${retryCount + 1}) for target language: ${channelName}`);
         retryCount++;
         setTimeout(() => {
-          publishToSession(channelName, publisher, session);
+          publishToSession(channelName, publisher, session, recordLangPublisher);
         }, RETRY_DELAY);
       } else {
         console.error('Maximum number of retries reached. Aborting publication.');
       }
     } else {
+      if (recordLangPublisher) recordLangPublisher(channelName);
       console.log(`Successfully published to Vonage for target language: ${channelName}`);
       retryCount = 0; // Reset the retry count
```

Comment:
Instead of passing a callback argument `recordLangPublisher` we can just return `true or false` from this function call `createTokboxPublisher` or we can save this info in the global store.